core/src/main/scala/com.snowplowanalytics.elasticsearch/loader/ElasticsearchSinkApp.scala

```diff
@@ -93,21 +95,26 @@ trait ElasticsearchSinkApp {
   */
   def convertConfig(config: ESLoaderConfig): KinesisConnectorConfiguration = {
     val props = new Properties
+    config.queueConfig match {
+      case qc: Kinesis =>
+        props.setProperty(KinesisConnectorConfiguration.PROP_KINESIS_ENDPOINT,
+          qc.endpoint)
+        props.setProperty(KinesisConnectorConfiguration.PROP_APP_NAME,
+          qc.appName.trim)
+        props.setProperty(KinesisConnectorConfiguration.PROP_INITIAL_POSITION_IN_STREAM,
+          qc.initialPosition)
+        props.setProperty(KinesisConnectorConfiguration.PROP_MAX_RECORDS,
+          qc.maxRecords.toString)
+        // So that the region of the DynamoDB table is correct
+        props.setProperty(KinesisConnectorConfiguration.PROP_REGION_NAME,
+          qc.region)
+
+
+      case _ => throw new RuntimeException("Kinesis configuration is not valid")
```

Comment:
I think it's better to have the method return an `Either[String, KinesisConnectorConfiguration]` which we will then combine with other methods which return `Either` in a for comprehension.

For an example (which uses `Validation` but that doesn't matter much) you can look at:

https://github.com/snowplow/snowplow/blob/9b67ff48fbdc6dde0fa11dfd7653a8eed4196647/3-enrich/stream-enrich/pubsub/src/main/scala/com.snowplowanalytics.snowplow.enrich.stream/sources/GooglePubSubSource.scala#L48-L68